### PR TITLE
[BTS-1583] Don't request search-alias view from cluster info second time

### PR DESCRIPTION
**Pull request by MBkkt**
### Scope & Purpose

- [x] :hankey: Bugfix
- [ ] :pizza: New feature
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [ ] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [ ] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [ ] Backport for 3.11: *(Please link PR)*
  - [ ] Backport for 3.10: *(Please link PR)*

#### Related Information

*(Please reference tickets / specification / other PRs etc)*

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [ ] GitHub issue / Jira ticket:
- [ ] Design document:

### CODE REVIEW

File: arangod/RestHandler/RestViewHandler.cpp [hunk 1]
```diff
@@ -262,175 +263,98 @@ void RestViewHandler::createView() {
...
-  if (!res.ok()) {
-    generateError(res);
-
-    return;
+  // ensure have the latest definition, by id because name is cached
+  if (view->type() != ViewType::kSearchAlias) {
+    auto newView = resolver.getView(view->id());
+    if (!newView) {
+      LOG_TOPIC("f58dc", WARN, Logger::CLUSTER)
+          << "Load plan for some strange reason doesn't contain this view";
```
Comment (Dronplane) [inline arangod/RestHandler/RestViewHandler.cpp:344]:
```suggestion
          << "Failed to refresh view definition from the Agency";
```

File: arangod/RestHandler/RestViewHandler.cpp [hunk 2]
```diff
@@ -262,175 +263,98 @@ void RestViewHandler::createView() {
...
-      view->properties(updated, LogicalDataSource::Serialization::Properties);
-
-  updated.close();
-
-  if (!res.ok()) {
-    generateError(res);
-
-    return;
+  // ensure have the latest definition, by id because name is cached
+  if (view->type() != ViewType::kSearchAlias) {
```
Comment (Dronplane) [inline arangod/RestHandler/RestViewHandler.cpp:340]:
```suggestion
  // Search-alias views are modified in place so we can avoid re-reading.
  if (view->type() != ViewType::kSearchAlias) {
```

**Review by Dronplane — APPROVED** [review]
LGTM

File: arangod/VocBase/LogicalView.cpp [hunk 3]
```diff
@@ -198,18 +200,18 @@ Result LogicalView::instantiate(LogicalView::ptr& view, TRI_vocbase_t& vocbase,
 }
 
 Result LogicalView::rename(std::string&& newName) {
+  if (!ServerState::instance()->isSingleServer()) {
+    return {TRI_ERROR_CLUSTER_UNSUPPORTED};
+  }
+  // TODO thread unsafe
```
Comment (nikita-vanyasin) [inline arangod/VocBase/LogicalView.cpp:206]: @MBkkt I'm not sure what is the policies for leaving TODOs in arangodb code, just checking if this is expected to be fixed someday? maybe some ticket was created?

Comment (MBkkt) [reply arangod/VocBase/LogicalView.cpp:206]: It was, I just fixed race for cluster

**Pull request merged**